### PR TITLE
fix: structs as cel env var

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

**Comment by adityathebe** [comment]
Some checks add data (cel's environment variables) as struct
- https://github.com/flanksource/canary-checker/blob/master/checks/exec.go#L74
- https://github.com/flanksource/canary-checker/blob/master/checks/aws_config.go#L60
- https://github.com/flanksource/canary-checker/blob/master/checks/folder_sftp.go#L48

and those cannot be converted to ref.Val.

**Comment by adityathebe** [comment]
This doesn't seem to work because the env can also contain durations which cannot be JSON marshalled

**Comment by moshloop** [comment]
@adityathebe  - can we not selectively go through the environment and convert only structs to maps, leaving the native types, time and duration as is ?

File: template.go [hunk 1]
```diff
@@ -129,3 +125,35 @@ func LoadSharedLibrary(source string) error {
...
+	for k, v := range in {
+		if reflect.ValueOf(v).Kind() != reflect.Struct {
+			newMap[k] = v
+			continue
+		}
+
+		var vMap map[string]any
+		dec, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{TagName: "json", Result: &vMap})
+		if err != nil {
+			fmt.Printf("error creating new mapstructure decoder: %v\n", err)
```
Comment (moshloop) [inline template.go:146]: I think these should be errors that are returned

**Review by moshloop — CHANGES_REQUESTED** [review]

**Pull request merged**